### PR TITLE
Refactor/error handling

### PR DESCRIPTION
**Pull request by tylrs**
## Description

This PR refactors previous  error handling in the following ways:
- Conditionally renders back button on error component to only show when necessary. For instance, if the first fetch call does not work, no back button is rendered, just a message telling the user to refresh the page. 
- Adds a switch statement for routes on App so that the very last Route will render if no matching path is found. This is the 404 error page
- Refactored conditionals in Route for displaying error and loading messages from if else if statements into double ampersand statements 

## Concerns

-  Any bugs still present? No bugs. Tests are updated to reflect changes
-  Next steps?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactor

## Checklist:
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: cypress/integration/homepage_spec.js [hunk 1]
```diff
@@ -185,9 +186,9 @@ describe('Homepage User Flows', () => {
           }
         }) 
         .visit('http://localhost:3000')
-        .get('a[href="/694919"]')
+        .get('a[href="/movies/694919"]')
```
Comment (andrewvallejo) [inline cypress/integration/homepage_spec.js:189]: Clever figuring out having to add movie for the switch statement

File: src/components/app/App.js [hunk 2]
```diff
@@ -54,35 +64,39 @@ class App extends Component {
...
-              unselectMovie={this.unselectMovie}
-              id={id}
-            />
-          }
-        }}/>
+        <Switch>
+          <Route exact path='/' render={() => {
+              return (
+                <>
+                  {this.state.error && <Error error={this.state.error} />}
```
Comment (andrewvallejo) [inline src/components/app/App.js:71]: Nice! Way to figure out how refactor this

File: src/components/app/App.js [hunk 3]
```diff
@@ -54,35 +64,39 @@ class App extends Component {
...
+                  movieInfo={this.state.selectedMovie} 
+                  selectMovie = {this.selectMovie}
+                  unselectMovie={this.unselectMovie}
+                  id={id}
+                />}
+              </>
+            )
+            }}
+          />
+          <Route path=''render={() => <Error error='404 Not Found' leaveError={this.leaveError}/>}/>
```
Comment (andrewvallejo) [inline src/components/app/App.js:98]: Do you think you need that path props? I would be curious if you could omit that

File: cypress/integration/homepage_spec.js [hunk 4]
```diff
@@ -276,10 +277,93 @@ describe('Homepage User Flows', () => {
...
         })
         cy.visit('http://localhost:3000')
-        .get('a[href="/694919"]')
+        .get('a[href="/movies/694919"]')
         .click()
         cy.contains('Could not retrieve selected movie, please try again')
+        cy.get('button').should('be.visible')
     });
+    it('Should show an error if user visits a bad url', () => {
+      cy.intercept('GET', 'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
```
Comment (andrewvallejo) [inline cypress/integration/homepage_spec.js]: Do we need this if the second intercept already does what it needs to?

**Review by andrewvallejo — CHANGES_REQUESTED** [review]

**Comment by tylrs** [comment]
Realized the test was making an actual API call accidentally instead of intercepting it. The test worked either way, but it worked unintentionally initially for the bad URL test

**Review by andrewvallejo — APPROVED** [review]

**Pull request merged**